hkmc2/shared/src/test/mlscript/codegen/BadNew.mls

```diff
@@ -11,14 +11,14 @@ new
 new 2
 //│ ═══[RUNTIME ERROR] TypeError: 2 is not a constructor
 
+// TODO: somehow desugaring lambdas fixed this, is this intended?
```

Comment:
The problem was that `new (...args) => ...` in JS is not valid syntax. But `new lambda` is fine. The fix would be to add parentheses around such new-ed arguments. But since lambdas should no longer be values, it probably doesn't matter much.